### PR TITLE
expose itemKey

### LINKED ISSUE #32: [feature] expose itemKey function
The `Repeats` mixin keeps track of the created DOM in the key/child map `_keyToChild`. When a child is not found, it will invoke `newChild` to create it and stores it in the `_keyToChild` map.
In this example `newChild` is invoked only for the newly added item, while `updateChild` is invoked for all the items:
```js
list.items = list.items.concat({name: 'new item'});
```

`_keyToChild` uses index as the key. This can be customized by providing the `itemKey` function to return a custom key `itemKey(item: any) => string|number`

Should we expose this property in `<virtual-list>`?

### PR DESCRIPTION
**Pull request by valdrinkoshi**
Fixes #32 

Added demo showing how using `itemKey` can affect rendering and animation
sorting-contacts

### CODE REVIEW

**Comment by valdrinkoshi** [comment]
@domenic PTAL, in particular the explainer part (README.md)

@graynorton FYI

File: README.md [hunk 1]
```diff
@@ -150,6 +158,57 @@ list.requestReset();
...
+```
+This renders 3 contacts, and the `<virtual-list>` key/Element map is:
+```
+0: <div>A</div>
+1: <div>B</div>
+2: <div>C</div>
+```
+We want to move the first contact to the end:
+```js
+const moveFirstContactToEnd = () => {
```
Comment (domenic) [inline README.md]: Nit: I would suggest using normal `function` syntax here.

File: README.md [hunk 2]
```diff
@@ -63,6 +63,14 @@ This is often used for node-recycling scenarios, as seen in [the example below](
 
 _We are discussing the naming and API for this functionality in [#25](https://github.com/valdrinkoshi/virtual-list/issues/25)._
 
+### `itemKey` property
+
+Type: `function(child: Element) => string|number`
```
Comment (domenic) [inline README.md]: string|number is strange. We should state how this is actually sorted. What if I mix strings and numbers? Does this just use JS's `array.sort()`?

Comment (valdrinkoshi) [reply README.md]: `itemKey` doesn't affect the sorting directly, as it just provides an identifier for a given item. `Repeats` uses this identifier to figure out if we should generate the DOM or not (if to invoke `newChild`)

Comment (domenic) [reply README.md]: Oh, right, that makes sense.

So why does it have to be a string or number? Couldn't it be anything?

Comment (valdrinkoshi) [reply README.md]: ...indeed! Will update the signature here :)

Comment (valdrinkoshi) [reply README.md]: What's an efficient way to say that it should be defined in the signature?
` => any` makes it look like it can be null or undefined

Comment (domenic) [reply README.md]: I think `=> any` is correct. If you return `null` or `undefined` that's bad in the same way that always returning `1` is bad.

Eventually we'll need to formally specify what happens if you do bad things, but for now we can just document the happy path.

Comment (valdrinkoshi) [reply README.md]: sg. Also, I realized that the input was incorrect, the function doesn't receive `Element` but the data itself...Corrected!

**Pull request merged**